storage/watcher.go

```diff
@@ -49,13 +49,24 @@ type WatchStream interface {
 
 	// Close closes the WatchChan and release all related resources.
 	Close()
+
+	// Rev returns the current revision of the KV the stream watches on.
+	Rev() int64
 }
 
+// WatchResponse contains the WatchID that can be used to cancel Watch,
+// and events that were watched, and Revision for clients to check the progress
+// of the Watch operation. If the Revision in the WatchResponse is same as the one
+// in the kv notification, it means that this watcher is in sync. If the Revision
+// in the WatchResponse is greater the one in kv notifiction, it means the watch is
+// unsynced; kv store is slow and not catching up with Watch with lower
 type WatchResponse struct {
 	// WatchID is the WatchID of the watcher this response sent to.
 	WatchID WatchID
 	// Events contains all the events that needs to send.
 	Events []storagepb.Event
+	// Revision is the latest revision of the storage.
```

Comment:
Revision is the revision of the KV when the watchResponse is created. For a normal response, the revision should be the same as the last modified revision inside Events. For a delayed response to a unsynced watcher, the revision is greater than the last modified revision inside Events.